### ISSUE TITLE
Future direction: literal type parameters

### ISSUE THREAD

**Issue by jackfoxy** [jackfoxy/DependentTypes#3]
This discussion https://github.com/robkuz/robkuz.github.io/issues/6#issuecomment-339061345 suggests the LimitedValue technique may be leveraged as an F# language suggestion to implement literals as type parameters. If this were to work out, it could potentially eliminate the need for the ````config```` consuming helper function, and open the door to other methods of creating dependent types.

**Comment by robkuz** [comment]
Yeah the design of this small library could be made even much simpler if the following [language suggestions](https://github.com/fsharp/fslang-suggestions/issues/608) would get implemented

**Comment by jackfoxy** [comment]
Also refer to this language suggestion https://github.com/fsharp/fslang-suggestions/issues/631
